src/io/ioexternalbus.h

```diff
@@ -37,13 +40,25 @@ class IOExternalBus : public IOBase {
     virtual void ioWrite(AInt offset, VInt value, unsigned size) override;
 
 protected:
-    virtual void parameterChanged(unsigned) override;
+    virtual void parameterChanged(unsigned) override{/* no parameters */};
+
+private slots:
+    void connectButtonTriggered();
 
 private:
+    bool m_Connected = false;
+    uint32_t m_ByteSize;
+    int32_t send_cmd(const uint32_t cmd, const uint32_t payload_size = 0, const QByteArray& payload = {});
+    int32_t recv_cmd(VBUS::CmdHeader& cmd_header);
+    int32_t recv_payload(QByteArray& buff, const uint32_t payload_size);
+    void disconnectOnError(QString msg = "");
```

Comment:
can we pass by const reference here? (`const QString&`).

Reply:
Yes.